templates/fine-tune-llm_v2/README.ipynb

```diff
@@ -145,73 +199,33 @@
-    "There, you'll also find some guidance on the LoRA parameters and why, in most cases, you don't need to change them.\n",
-    "\n",
-    "### I have the right model, context length and everything. Can I optimize compute cost?\n",
-    "\n",
-    "Optimizing your fine-tuning runs for compute cost is a non-trivial problem.\n",
-    "The default configs in this template require the following compute:\n",
-    "Llama-3-8B and Mistral require 16 A10Gs. Llama-3-70B and Mixtral require 32 A10Gs.\n",
-    "\n",
-    "Before optimizing for compute, make sure that you have selected a context length that is long enough for your dataset. If you have very few datapoints in your dataset that requires a much larger context than the others, consider removing them. The model of your choice and fine-tuning technique should also suit your data.\n",
-    "\n",
-    "If you want different compute, we *suggest* the following workflow to find a suitable configuration:\n",
-    "\n",
-    "* Start with a batch size of 1\n",
-    "* Choose a GPU instance type that you think will give you good flops/$. If you are not sure, here is a rough guideline:\n",
-    "    * g5 nodes for high availability\n",
-    "    * p4d/p4de nodes for lower availability but better flops/$\n",
-    "    * Anything higher-end if you have the means of acquiring them\n",
-    "* Do some iterations of trial and error on instance types and deepspeed settings to fit the workload while keeping other settings fixed\n",
-    "    * Use deepspeed stage 3 (all default configs in this template use stage 3)\n",
-    "    * Try to use deepspeed offloading only if it reduces the minimum number of instances you have to use\n",
-    "        * Deepspeed offloading slows down training but allows for larger batch sizes because of a more relaxed GRAM foot-print\n",
-    "    * Use as few instances as possible. Fine-tune on the same machine if possible.\n",
-    "        *  The GPU to GPU communication across machines is very expensive compared to the memory savings it could provide. You can use a cheap CPU-instance as a head-node for development and a GPU-instance that can scale down as a worker node for the heavy lifting.\n",
-    "        * Training single-node on A100s may end up cheaper than multi-node on A10s if availablity is not an issue\n",
-    "* Be aware that evaluation and checkpointing introduce their own memory-requirements\n",
-    "   * If things look good, run fine-tuning for a full epoch.\n",
-    "* After you have followed the steps above, increase batch size as much as possible without OOMing.\n",
-    "\n",
-    "We do not guarantee that this will give you optimal settings, but have found this workflow to be helpful ourselves in the past.\n",
-    "\n",
-    "### I've reviewed the customizable hyperparameters available. How can I get even more control?\n",
```

Comment:
Isn't this "How can I get even more control?" better left in? Important disclaimer for people who know what they're doing and want research-like flexibility.

Reply:
I think it's fine since its redundant. I want to trim off any redundant info. This makes it much easier to update content in the future.